[thaddeusong] iP

thaddeusong opened the pull request:
## Duke Project

> Beware the stories you read or tell; subtly, at night, beneath the waters of consciousness, they are altering your world. - [Ben Okri](https://en.wikipedia.org/wiki/Ben_Okri)

This project is
- **Text**-based
- *Easy* to use

This project helps you to:

1. Track tasks
2. Stay on target
3. Practise programming
:+1:

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

> If you Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Eclipse-Dominator committed:
Add Gradle Support

bladedmistral committed:
Level-0

bladedmistral committed:
Level-1

bladedmistral committed:
Level-2

bladedmistral committed:
Level-3 and Level-4

bladedmistral committed:
Add Testing

bladedmistral committed:
Add Exceptions

bladedmistral committed:
Add Delete method

bladedmistral committed:
Use More OOP

(cherry picked from commit 735bfbe59b74709b7cea91ab32e9e914a9c4c062)

bladedmistral committed:
Add File Saving

bladedmistral committed:
Reorganize Files

bladedmistral committed:
Add JUnit Tests to Deadline and Todo

bladedmistral committed:
Add File Saving

bladedmistral committed:
Add LocalDateTime representation

bladedmistral committed:
Refactor Class Names

(cherry picked from commit b6f226b0e3c678c3e35d5683bc5584777ed446dc)

bladedmistral committed:
Refactor Class Names

bladedmistral committed:
Fix Task Constructors

bladedmistral committed:
Fix Task Constructors

(cherry picked from commit 2045c3cfd822ef9877bea3c41264951426d37a54)

bladedmistral committed:
Add JavaDoc

bladedmistral committed:
Add Unversioned Files

bladedmistral committed:
Add Unversioned Files

bladedmistral committed:
Follow Coding Standard

bladedmistral committed:
Add Find Command

bladedmistral committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Duke/application/Application.java
#	src/main/java/Duke/command/Command.java
#	src/main/java/Duke/message/Message.java
#	src/main/java/Duke/task/Deadline.java

bladedmistral committed:
Resolve Merge Conflicts

bladedmistral committed:
Resolve Merge Conflicts

thaddeusong committed:
Update README.md

bladedmistral committed:
Fix Style

bladedmistral committed:
Fix Style

bladedmistral committed:
Fix README Conflict

bladedmistral committed:
Add JavaFX GUI

bladedmistral committed:
Add Unversioned Files

bladedmistral committed:
Merge branch 'branch-Level-10'

andrechuakj commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,232 @@
+    protected Delete(String content) {
+        super(content);
+    }
+    @Override
```
Perhaps adding a newline before the method below would make it clearer?

andrechuakj commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,131 @@
+ * Handles user interactions, input parsing, and displaying messages.
+ */
+public class Ui {
+    private static final double GUI_WIDTH = 800.0;
+    private static final double GUI_HEIGHT = 600.0;
```
I like the use of consistent naming for the constants here. 👍

andrechuakj commented on src/main/java/duke/application/Application.java:
```diff
@@ -0,0 +1,149 @@
+ * The application that manages UI and the taskList.
+ */
+public class Application extends javafx.application.Application {
+    private static final String name = "Iris";
+    private static final String filePath = "C:\\Users\\ortt2\\Documents\\ip\\src\\data\\tasks.txt";
```
Would changing these constants to full uppercase be clearer?

wr1159 replied:
Perhaps changing the naming to use all uppercase with underscore would follow the coding standard better?

```suggestion
    private static final String NAME = "Iris";
    private static final String FILE_PATH = "C:\\Users\\ortt2\\Documents\\ip\\src\\data\\tasks.txt";
```

andrechuakj commented on src/main/java/duke/application/Application.java:
```diff
@@ -0,0 +1,149 @@
+     * @param content The substring to search for.
+     * @return A TaskList containing matching Tasks.
+     */
+    public TaskList findMatchingTasks(String content) {
```
I like that the method name is descriptive which makes it easy to understand what this method does.

andrechuakj reviewed (commented):
LGTM. Well done!

wr1159 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,232 @@
+     */
+    public static Command of(String commandName, String content)
+            throws NoCommandFoundException {
+        switch (commandName) {
+        case "bye":
+            return new Bye(content);
+        case "todo":
+            return new Todo(content);
+        case "deadline":
+            return new Deadline(content);
+        case "event":
+            return new Event(content);
+        case "find":
+            return new Find(content);
+        case "list":
+            return new List(content);
+        case "mark":
+            return new Mark(content);
+        case "unmark":
+            return new Unmark(content);
+        case "delete":
+            return new Delete(content);
+        default:
+            throw new NoCommandFoundException(commandName);
+        }
```
I like that you followed the coding standard for case and switch which is a common mistake!

wr1159 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,232 @@
+class Bye extends Command {
+    protected Bye(String content) {
+        super(content);
+    }
+    @Override
```
Perhaps there can be an extra space in between the bracket and the `@Override`. I noticed the same issue with the following functions too.

```suggestion
    }
    
    @Override
```

wr1159 commented on src/main/java/duke/message/Message.java:
```diff
@@ -0,0 +1,158 @@
+     */
+    public Message(String content) {
+        this.content = content;
+        //Set label contents and formatting
```
Perhaps there could be an empty space between `//` and `Set` to make it clearer? I notice the same issue with the following lines.
```suggestion
        // Set label contents and formatting
```

wr1159 commented on src/main/java/duke/application/Application.java:
```diff
@@ -0,0 +1,149 @@
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.ui.Ui;
+import javafx.application.Platform;
+import javafx.stage.Stage;
```
Perhaps there should be a new line in between for clarity in importing?

```suggestion
import duke.ui.Ui;

import javafx.application.Platform;
import javafx.stage.Stage;
```

wr1159 commented on src/main/java/duke/application/Application.java:
```diff
@@ -0,0 +1,149 @@
+package duke.application;
+
```
Maybe this line could be removed if it is redundant? 
```suggestion
```

wr1159 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,90 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+
+import duke.exception.DukeException;
+import duke.exception.InvalidTaskFormatException;
+import duke.exception.InvalidTimeFormatException;
```
I like the separation of import here, it is very clean to read! 👍

wr1159 reviewed (commented):
Overall looks good to me! I mostly left comments about spacing and they are all very quick fixes. Good job on maintaining this clean code!

bladedmistral committed:
Fix Image URL Sourcing

Sourcing the image from the URL rather than storing it as a local file makes it easier to change the image in the future.

bladedmistral committed:
Add Assertions to TaskList

bladedmistral committed:
Remove magic literals from UI class

Added dimensions of various GUI elements as static
fields, so that they can be sacled appropriately.

thaddeusong committed:
Merge pull request #1 from thaddeusong/branch-A-CodeQuality

Remove magic literals from UI class

bladedmistral committed:
Add Duplicate detection

Prevents tasks from being added if the name is identical to a previous task.

bladedmistral committed:
Make Error Messages Red

bladedmistral committed:
Make Error Messages Red

bladedmistral committed:
Add Ui.png

bladedmistral committed:
Update User Guide

bladedmistral committed:
Put Ui.png in base folder
